com.dropbox.Client.json

```diff
@@ -12,7 +12,16 @@
         "--socket=pulseaudio",
         "--share=network",
         "--filesystem=home",
-        "--filesystem=/tmp"
+        "--filesystem=/tmp",
+	/* StatusNotifierItem support */
+	"--talk-name=org.kde.StatusNotifierWatcher",
+	"--own-name=org.kde.StatusNotifierItem-7-1"
+        /* Note: the number 7 above is the PID of the dropboxd process
```

Comment:
Ugh, I don't think this is meaningfully useful. Sadly the StatusNotifier Spec isn't very good in this area and I think you just have to give full `--own-name=org.kde.*` permissions which really really sucks.

Reply:
Honestly a discussion just needs to happen with the KDE folks about improving the SNI spec.

- PID is not a reliable unique identifier
- Owned names should be namespaced under the application, not under KDE

Reply:
Actually reading the linked spec applications can pass full names so you probably don't have to be under that namespace in practice. I also question the ability for all applications to be able to register being a StatusNotifierHost.

Reply:
I'll try to raise the issue with KDE, if I find the right contacts. Ubuntu's `libappindicator` [3] reuses the process' own connection name (beginning with a `:`), which is guaranteed to be unique by the D-Bus spec [4], and it uses instead different object names to register multiple indicators. The way it works is explained in [2], but apparently it never made it to the freedesktop.org spec. For apps that register SNI's in this way, it suffices to add `--talk-name=org.kde.StatusNotifierWatcher` to have working indicators. But of course it can't be fixed here unless Dropbox changes its implementation.

Shall I close the pull request?

[3] https://bazaar.launchpad.net/~indicator-applet-developers/libappindicator/trunk.16.10/view/head:/src/app-indicator.c#L1387
[4] https://dbus.freedesktop.org/doc/dbus-specification.html#term-unique-name

Reply:
I think this conversation is worth keeping open and honestly still might be worth merging. There isn't a security cost (we have `home`) so when it works thats useful.

Reply:
I have done some further tests, but it seems like `--own-name=org.kde.*` is the only reasonable approach. It is buggy: the "Recently Changed Files" menu stays empty, and one needs to double click on the icon at least once to force the menu to populate. After that, it seems to work. A double click on the icon does bring up the normal, Dropbox-rendered menu, so there is no loss of functionality with respect to the systray (apart from the risk of different programs running with the same PID).

An alternative could be adding `libappindicator-gtk*` from [https://github.com/flathub/shared-modules](url) (both gtk2 and gtk3 work) and the option `--env=XDG_CURRENT_DESKTOP=Unity`, but that does not work at all. Dropbox must rely on some peculiarities of Unity that are not present in other desktops: with `libappindicator` and the GNOME Shell extension (K)StatusNotifierItem/Appindicator Support, it fails to show the icon with the error:
`[...] LIBDBUSMENU-GLIB-WARNING **: [...] About to Show called on an item wihtout submenus. We're ignoring it.`
I have found some reports of this elsewhere [4], [5]. The solution from `nautilus-dropbox` has been to isolate Dropbox from D-Bus entirely [6], thus falling back to the systray.

On top of that, it seems that Dropbox still uses its PID as unique id when running with `libappindicator` (it registers the SNI at `/org/ayatana/NotificationItem/dropbox_client_PID`).

[4] https://www.dropboxforum.com/t5/Installation-and-desktop-app/dropbox-installation-hard-problem/td-p/266081
[5] https://bugs.launchpad.net/ubuntu/+source/nautilus-dropbox/+bug/1546176
[6] https://github.com/dropbox/nautilus-dropbox/pull/49

Reply:
Yea looks like they have a custom implementation.